parser/ParseStatement.hpp

```diff
@@ -195,22 +197,55 @@ class ParseStatementCreateIndex : public ParseStatement {
      *                            If specified as null, then index is created
      *                            on all the attributes.
      * @param index_type The type of index to create.
-     * @param index_property_list A list of properties for the index that modify
-     *                            the default properties of the index.
      **/
     ParseStatementCreateIndex(const int line_number,
                               const int column_number,
                               ParseString *index_name,
                               ParseString *relation_name,
-                              PtrList<ParseString> *attribute_name_list,
-                              ParseString *index_type,
-                              PtrList<ParseKeyValue> *index_property_list)
+                              PtrList<ParseAttribute> *attribute_list,
+                              ParseString *index_type)
       : ParseStatement(line_number, column_number),
         index_name_(index_name),
         relation_name_(relation_name),
-        attribute_name_list_(attribute_name_list),
-        index_type_(index_type),
-        index_property_list_(index_property_list) {
+        attribute_list_(attribute_list),
+        index_type_(index_type) {
+          initializeIndexType();
+    }
+
+    /**
+     * @brief Constructor.
+     *
+     * @param index_name The name of the index to create.
+     * @param relation_name The name of the relation to create index upon.
+     * @param attribute_name_list A list of attributes of the relation
+     *                            on which the index has to be created.
+     *                            If specified as null, then index is created
+     *                            on all the attributes.
+     * @param index_type The type of index to create.
+     * @param index_properties_line_number
+     * @param index_properties_column_number
+     * @param index_properties
+     * TODO(ssaurabh) Fix comments
+     **/
+    ParseStatementCreateIndex(const int line_number,
+                              const int column_number,
+                              ParseString *index_name,
+                              ParseString *relation_name,
+                              PtrList<ParseAttribute> *attribute_list,
+                              ParseString *index_type,
+                              const int index_properties_line_number,
```

Comment:
As a suggestion, it might make it easier to resolve index properties if they were abstracted into a small helper class, `IndexProperties` which would essentially be a wrapper around the properties. It could do things like check for duplicate properties, or lookup a property with a given key. 

I found it to be helpful to do it this way in the `BlockProperties` feature